pkg/hyper/network.go

```diff
@@ -17,140 +17,70 @@ limitations under the License.
 package hyper
 
 import (
-	"fmt"
-	"net"
+	"strings"
 
 	"github.com/golang/glog"
 
 	"github.com/containernetworking/cni/pkg/ns"
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
 	"github.com/vishvananda/netlink"
 )
 
-type NetNsInfos struct {
-	Interfaces []*InterfaceInfo
+type NetworkInfo struct {
+	IfName     string
+	Mac        string
+	Ip         string
+	Gateway    string
+	BridgeName string
 }
 
-type InterfaceInfo struct {
-	Ip      string
-	Gateway string
-	Mac     string
-}
-
-func getNetNsInfos(netns ns.NetNS) *NetNsInfos {
-	result := &NetNsInfos{}
-
-	var infos []*InterfaceInfo
-	netns.Do(func(_ ns.NetNS) error {
-		infos = collectInterfaceInfos()
-		return nil
-	})
-	if len(infos) != 0 {
-		result.Interfaces = infos
-	}
-
-	return result
-}
+func convertCNIResult2NetworkInfo(result cnitypes.Result) *NetworkInfo {
+	ret := &NetworkInfo{}
 
-func collectInterfaceInfos() []*InterfaceInfo {
-	infos := []*InterfaceInfo{}
-
-	links, err := netlink.LinkList()
+	r, err := current.GetResult(result)
 	if err != nil {
-		return infos
+		glog.Errorf("Convert CNI Result failed")
+		return nil
 	}
 
-	for _, link := range links {
-		if link.Type() == "lo" {
-			// only omit "lo" here
+	// only handle bridge plugin now
+	for _, iface := range r.Interfaces {
+		// veth in host
+		if strings.Contains(iface.Name, "veth") {
```

Comment:
Doesn't this check really required? I think `iface.Sandbox != ""` is enough for getting container interfaces